src/pybind/mgr/dashboard/frontend/src/app/shared/classes/cd-helper.class.ts

```diff
@@ -0,0 +1,19 @@
+import * as _ from 'lodash';
+
+export class CdHelperClass {
+  /**
+   * Simple way to only update variables on change to reduce draw cycles
+   *
+   * @param componentThis - In order to update the variables if necessary
+   * @param change - The variable name (attribute of the object) is followed by the current value
+   *                 it would update even if it equals
+   */
+  static updateChanged(componentThis: any, change: { [publicVarName: string]: any }) {
+    Object.keys(change).forEach((publicVarName) => {
+      const data = change[publicVarName];
+      if (!_.isEqual(data, componentThis[publicVarName])) {
+        componentThis[publicVarName] = data;
+      }
+    });
+  }
+}
```

Comment:
This function allows you to easily replace changed data. It's very useful if you don't have an Observable. You can do your post processing first and than compare with the current variable, if there is a change it get's replaced in your component. I see a lot of potential for refactorings (which aren't included in this PR).

It's not reinventing the wheel it's just a neat little function to update on change. The structure is simple that you have to provide and is more flexible. Currently it's only used two times but that's because I have not refactored the whole dashboard yet with this function ;) as their are countless places where it would be a good fit or where something similar is already done, but I would love to not add this change into this PR and put the change of it in a follow up PR as it will take some time to find all places.